lib/spack/spack/spec.py

```diff
@@ -2129,10 +2129,12 @@ def format(self, format_string='$_$@$%@+$+$=', **kwargs):
             ${OPTIONS}       Options
             ${ARCHITECTURE}  Architecture
             ${SHA1}          Dependencies 8-char sha1 prefix
+            ${HASH:len}      DAG hash with optional length specifier
 
             ${SPACK_ROOT}    The spack root directory
             ${SPACK_INSTALL} The default spack install directory,
                              ${SPACK_PREFIX}/opt
+            ${PREFIX}        The package prefix
```

Comment:
I tried to search for uses of full-string substitutions in the whole code base. Am I wrong in saying that there's none ?